dulwich/config.py

```diff
@@ -143,7 +146,11 @@ def get_boolean(self, section, name, default=None):
             return True
         elif value.lower() == b"false":
             return False
-        raise ValueError("not a valid boolean string: %r" % value)
+
+        if strict:
+            raise ValueError("not a valid boolean string: %r" % value)
+        else:
+            return value
```

Comment:
This makes get_boolean() return values that are not boolean. Please just use Config.get() in that case.